spring-cloud-dataflow-apps-plugin/spring-cloud-dataflow-apps-generator-plugin/src/main/java/org/springframework/cloud/dataflow/app/plugin/MavenXmlWriter.java

```diff
@@ -103,32 +103,37 @@ public static String toXml(Plugin plugin) {
 			throw new IllegalStateException("Issue creating config for additional plugin", ex);
 		}
 
-		return write(serializer -> invokeMavenXppWriteMethod(
+		return write("plugin", serializer -> invokeMavenXppWriteMethod(
 				plugin, "writePlugin", "plugin", serializer));
 	}
 
 	/**
-	 * Template method that helps generating well-formed and valid XML segments. It uses the {@link XmlSerializer}
-	 * to create the create the XML structure and delegates to the #elementWriter callback to fill in the content.
+	 * Generates well-formed and valid XML segments with the help of
+	 * {@link XmlSerializer}.
 	 *
-	 * @param elementWriter Callback handler that contributes the XML content.
-	 * @return Returns well-formed XML element.
+	 * @param xmlTagName the tag of the element
+	 * @param elementWriter callback handler that writes the XML content
+	 * @return well-formed XML element with namespace element removed
 	 */
-	public static String write(Consumer<XmlSerializer> elementWriter) {
+	public static String write(String xmlTagName, Consumer<XmlSerializer> elementWriter) {
 		try {
 			Writer writer = new StringWriter();
-
 			XmlSerializer serializer = new MXSerializer();
 			serializer.setProperty("http://xmlpull.org/v1/doc/properties.html#serializer-indentation", "  ");
 			serializer.setProperty("http://xmlpull.org/v1/doc/properties.html#serializer-line-separator", "\n");
 			serializer.setOutput(writer);
-
 			serializer.startDocument("UTF-8", null);
 			elementWriter.accept(serializer);
 			serializer.endDocument();
-
 			String result = writer.toString();
-			return result.substring(result.indexOf('\n') + 1); // remove first line (e.g. remove the <?xml ... ?>)
+			// Strip the namespace info
```

Comment:
Latest version of `MXSerializer` handles generated xml a bit different. This impl has been improved to look for the actual tag, not the 1st newline character for parsing.